apps/hannk/Makefile

```diff
@@ -6,25 +6,52 @@ include ../support/Makefile.inc
 # builds but easier to enable it for everything.
 CXXFLAGS += -Wno-unused-private-field -fno-exceptions -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wunused-variable -I$(MAKEFILE_DIR)
 
+BENCHMARK_OUT = benchmark
+ifeq (hexagon-32-qurt,$(findstring hexagon-32-qurt,$(HL_TARGET)))
+	# Building benchmark application as shared object instead of elf for
+	# hexagon target.
+	BENCHMARK_OUT := $(BENCHMARK_OUT).so
+	BENCHMARK_HEXAGON_FLAGS = -shared -fPIC -G0
+	HEXAGON_STUBS = $(BIN)/$(HL_TARGET)/stubs.o
+endif
+
+
 ifneq (,$(findstring -O,$(OPTIMIZE)))
 	CXXFLAGS += -DNDEBUG
 endif
 
 MAKEFILE_DIR=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 
-.PHONY: all build clean test delegate
+.PHONY: all build clean test delegate test-hexagon-sim test-hexagon-device
 all: build
 
 # .SECONDARY with no prerequisites causes all targets to be treated as secondary
 # (i.e., no target is removed because it is considered intermediate).
 .SECONDARY:
 
 build: \
-	$(BIN)/$(HL_TARGET)/benchmark \
+	$(BIN)/$(HL_TARGET)/$(BENCHMARK_OUT) \
 	$(BIN)/$(HL_TARGET)/compare_vs_tflite
 
 test: compare_vs_tflite
-	$(BIN)/$(HL_TARGET)/compare_vs_tflite test/*/* --benchmark 0
+	$(foreach test_model, $(shell ls -1 test/*/*), $(BIN)/$(HL_TARGET)/compare_vs_tflite $(test_model) --benchmark 0;)
+
+test-hexagon-sim: $(BIN)/$(HL_TARGET)/$(BENCHMARK_OUT)
+	@mkdir -p $@
+	echo "Benchmarking tests..."
+	echo "$(HEXAGON_SDK_ROOT)/rtos/qurt/computev65/debugger/lnx64/qurt_model.so" > $(BIN)/$(HL_TARGET)/osam.cfg
```

Comment:
It looks like these constants could go in the run_hexagon_sim.sh script too?

Reply:
Done

Reply:
Sorry forgot to do these.